replica/table.cc

```diff
@@ -2782,9 +2782,10 @@ class compaction_group::table_state : public compaction::table_state {
     }
     future<> on_compaction_completion(sstables::compaction_completion_desc desc, sstables::offstrategy offstrategy) override {
         if (offstrategy) {
-            return _cg.update_sstable_lists_on_off_strategy_completion(std::move(desc));
+            co_await _cg.update_sstable_lists_on_off_strategy_completion(std::move(desc));
+            _cg.trigger_compaction();
```

Comment:
missing co_return now.

Reply:
Is it really needed? I was under the impression that co_return is optional if return type is ` future<> ` and we waited for all deferring tasks with co_await.Is it really needed? I was under the impression that co_return is optional if return type is future<> and we waited for all deferring tasks with co_await.

Reply:
I left it missing on purpose. I just noticed an example in the code base where co_return is also not specified, under the same conditions.

Reply:
previously the function returned in this block. Now, after `co_await _cg.update_sstable_lists_on_off_strategy_completion` and `_cg.trigger_compaction`
the function will happily continue and call `co_await _cg.update_main_sstable_list_on_compaction_completion`.
so you need to either `co_return` from the `if` block, or put the rest in `else {}`.